core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3585,14 +3585,17 @@ private void setAttributeSingleFile(RpcContext rpcContext, LockedInodePath inode
           protoOptions.hasReplicationMin() ? protoOptions.getReplicationMin() : null;
       mInodeTree.setReplication(rpcContext, inodePath, replicationMax, replicationMin, opTimeMs);
     }
-    if (protoOptions.hasTtl()) {
-      long ttl = protoOptions.getTtl();
-      if (inode.getTtl() != ttl || inode.getTtlAction() != protoOptions.getTtlAction()) {
+    if (protoOptions.hasCommonOptions() && protoOptions.getCommonOptions().hasTtl()
+        && protoOptions.getCommonOptions().hasTtlAction()) {
+      long ttl = protoOptions.getCommonOptions().getTtl();
+      if (inode.getTtl() != ttl
+          || inode.getTtlAction() != protoOptions.getCommonOptions().getTtlAction()) {
```

Comment:
Can these conditions be cleaned up, it seems like we check `inode.getTtl()` twice, if `entry.setTtl` is idempotent, we could just set it regardless of whether or not it has changed (since the main thing is whether or not to update last mod time).

Reply:
While minor, the tradeoff is whether or not checking `inode.getTtl` twice is worth the tradeoff of setting the field on the entry. It looks like under the covers the entry does some other iother internal actions when you set a field. I don't know what the overhead is like, but doing that boolean comparison could possibly save some operations rather than setting it every time. An alternative so we don't do the comparison twice could be 

```
      long ttl = protoOptions.getCommonOptions().getTtl();
      boolean changedTtl = inode.getTtl() != ttl;
      if (changedTtl
          || inode.getTtlAction() != protoOptions.getCommonOptions().getTtlAction()) {
        if (changedTtl) {
          entry.setTtl(ttl);
        }
        entry.setLastModificationTimeMs(opTimeMs);
        entry.setTtlAction(ProtobufUtils.toProtobuf(
            protoOptions.getCommonOptions().getTtlAction()));
      }
```

Reply:
My suggestion is for improving code readability and not performance. I don't think the cost of either checking the boolean or setting the fields should be significant.